Source/Core/Common/FileUtil.cpp

```diff
@@ -464,14 +464,12 @@ FSTEntry ScanDirectoryTree(const std::string& directory, bool recursive)
   {
     const std::string virtual_name(TStrToUTF8(ffd.cFileName));
 #else
-  struct dirent dirent, *result = nullptr;
-
   DIR* dirp = opendir(directory.c_str());
   if (!dirp)
     return parent_entry;
 
   // non Windows loop
-  while (!readdir_r(dirp, &dirent, &result) && result)
+  while (struct dirent* result = readdir(dirp))
```

Comment:
According to dirent.h man, dirent is defined as a struct not a typedef so that ```struct``` should be needed. Otherwise, it isn't standard and might fail to compile. Unless there is a rule in C/C++ standard that I'm unaware and allows to declare a struct without the struct keyword (I think there is one only if there is no ambiguity, but I'm not sure).

Can someone else confirm? @lioncash ?